### PR TITLE
Add legacy search params

### PR DESCRIPTION
**Pull request by zzacharo**

### CODE REVIEW

File: site/zenodo_rdm/params.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    all_versions = fields.Boolean()
+    communities = fields.List(fields.String())
+    status = fields.String()
+    type = fields.String()
+    subtype = fields.String()
+
+    @post_load
+    def dump_all_versions(self, data, **kwargs):
+        if "all_versions" in data:
+            data["allversions"] = data["all_versions"]
```
Comment (slint) [inline site/zenodo_rdm/params.py:101]: minor/question: does this take into account when just passing `?all_versions` (i.e. without any value)?

**Review by slint — APPROVED** [review]
LGTM, one comment on `?all_versions` behaviour

File: site/zenodo_rdm/params.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    all_versions = fields.Boolean()
+    communities = fields.List(fields.String())
+    status = fields.String()
+    type = fields.String()
+    subtype = fields.String()
+
+    @pre_load
+    def load_all_versions(self, data, **kwargs):
+        if "all_versions" in data:
+            if not data["all_versions"]:
```
Comment (zzacharo) [inline site/zenodo_rdm/params.py:101]: @slint now it is taken into account

Comment (slint) [inline site/zenodo_rdm/params.py:101]:
```suggestion
            if data["all_versions"] in (None, ""):
```

Maybe this is more safe against `False` or `0`, (or whatever Flask considers Falsy querystring params)?

**Pull request merged**